clients/fides-js/src/components/Overlay.tsx

```diff
@@ -49,15 +49,18 @@ const Overlay: FunctionComponent<Props> = ({
   const hasMounted = useHasMounted();
   const [bannerIsOpen, setBannerIsOpen] = useState(false);
 
-  const dispatchCloseEvent = useCallback(() => {
-    dispatchFidesEvent("FidesModalClosed", cookie, options.debug);
-  }, [cookie, options.debug]);
+  const dispatchCloseEvent = useCallback(
+    ({ saved = false }: { saved?: boolean }) => {
+      dispatchFidesEvent("FidesModalClosed", cookie, options.debug, { saved });
```

Comment:
we need to know if the modal was closed without saving. gpp signals that it is not ready when the modal opens. once it saves, it signals it is ready again. but if you close the modal without saving, we'd be stuck in a "not ready" state